src/users.ts

```diff
@@ -0,0 +1,27 @@
+import axios from 'axios'
+
+const ZORA_API_BASE_URL = 'https://zora.co'
+const MAX_USERS_PER_REQUEST = 100
+
+/**
+ * Returns a list of Zora user profiles given a list of up to 100 addresses
+ * @param addresses
+ */
+export async function getZoraProfiles(addresses: string[]) {
+  if (addresses.length === 0) {
+    throw new Error('Empty addresses array')
+  }
+  if (addresses.length > MAX_USERS_PER_REQUEST) {
+    throw new Error(`Addresses array exceeds max length of ${MAX_USERS_PER_REQUEST}`)
+  }
+  try {
+    const res = await axios.post(`${ZORA_API_BASE_URL}/api/users`, { addresses })
+    if (!res.data || !Array.isArray(res.data) || res.data.length === 0) {
+      throw new Error()
+    }
+    return res.data
+  } catch (err) {
+    const msg = err.message ? err.message : 'Error retrieving users'
```

Comment:
could we make the error message a bit more specific? 

Do the different failures on https://github.com/ourzora/zdk/pull/48/files#diff-e5913cca70de3baa9860b5af30dc3cc647849c63c76161f213cf7ec3bf2b464fR19 have different reasons on the api side?

Reply:
@sporkspatula check out the updated error handling. it will first look for a custom error message from the zora api, then will look for a standard Error.message, and finally will default to 'Error retrieving users'.